src/plugins/intel_gpu/tests/test_cases/reorder_gpu_test.cpp

```diff
@@ -1625,9 +1624,16 @@ TEST(reorder_gpu_opt, non_trivial_remove_redundant)
     auto executed_primitives = net.get_executed_primitives();
     auto all_primitives = net.get_all_primitives();
 
+    const auto& target_reorder = net.get_primitive("r1");
+    ASSERT_EQ(target_reorder->can_be_optimized(), false);
```

Comment:
Easy fix is to return when supports_immad.
Properly fix is to check other field than event.